components/ciphertool/src/main/java/org/wso2/ciphertool/cipher/AsymmetricCipher.java

```diff
@@ -0,0 +1,103 @@
+import org.wso2.ciphertool.exception.CipherToolException;
+import org.wso2.ciphertool.utils.Constants;
+import org.wso2.ciphertool.utils.KeyStoreUtil;
+import org.wso2.ciphertool.utils.Utils;
+
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.util.Base64;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+
+/**
+ * Provides methods for encryption and decryption using asymmetric key algorithms.
+ */
+public class AsymmetricCipher implements CipherMode {
+
+    private final String keyAlias;
+    private final KeyStore keyStore;
+    private final Cipher cipher;
+    private static final String CIPHER_INIT_ERROR_MESSAGE = "Error initializing Cipher.";
+    private static final String GET_KEY_ERROR_MESSAGE = "Error retrieving key associated with alias : ";
+
+    public AsymmetricCipher(KeyStore keyStore, String keyAlias) {
+
+        this.keyStore = keyStore;
+        this.keyAlias = keyAlias;
+        String cipherTransformation = System.getProperty(Constants.CIPHER_TRANSFORMATION_SYSTEM_PROPERTY);
+        String algorithm = StringUtils.isNotBlank(cipherTransformation)
+                ? cipherTransformation : Constants.RSA;
+        try {
+            this.cipher = Cipher.getInstance(algorithm);
+        } catch (NoSuchAlgorithmException | NoSuchPaddingException e) {
+            throw new CipherToolException(CIPHER_INIT_ERROR_MESSAGE, e);
+        }
+    }
+
+    public AsymmetricCipher(KeyStore keyStore) {
+
+        this(keyStore, System.getProperty(Constants.KEY_ALIAS_PROPERTY));
+    }
+
+    /**
+     * Encrypt plain text using asymmetric encryption.
+     *
+     * @param plainText Plain text password.
+     * @return Encrypted password.
+     */
+    @Override
+    public String doEncryption(String plainText) {
+
+        try {
+            Certificate certs = this.keyStore.getCertificate(this.keyAlias);
+            cipher.init(Cipher.ENCRYPT_MODE, certs);
+        } catch (KeyStoreException e) {
+            throw new CipherToolException(GET_KEY_ERROR_MESSAGE + this.keyAlias, e);
+        } catch (InvalidKeyException e) {
+            throw new CipherToolException("The provided public cert is invalid.", e);
+        }
+        return Utils.doEncryption(cipher, plainText);
+    }
+
+    @Override
+    public String doDecryption(String cipherText) {
+
+        try {
+            Key privateKey = this.keyStore.getKey(this.keyAlias, KeyStoreUtil.getKeystorePassword().toCharArray());
+            cipher.init(Cipher.DECRYPT_MODE, privateKey);
+        }  catch (NoSuchAlgorithmException | InvalidKeyException e) {
+            throw new CipherToolException("The provided private key is invalid.", e);
```

Comment:
You can collect all the error messages to a single error class enums. So the code is clear